x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/connector_detail/attach_index_box.tsx

```diff
@@ -65,66 +65,109 @@ export const AttachIndexBox: React.FC<AttachIndexBoxProps> = ({ connector }) =>
+  const isLoading = [Status.IDLE, Status.LOADING].includes(status);
+
+  // Helper function to remove the managed connector index prefix from the index name
+  const removePrefixConnectorIndex = (indexName: string) => {
+    if (!connector.is_native) {
+      return indexName;
+    }
+    if (indexName.startsWith(MANAGED_CONNECTOR_INDEX_PREFIX)) {
+      return indexName.substring(MANAGED_CONNECTOR_INDEX_PREFIX.length);
+    }
+    return indexName;
+  };
+
+  // Helper function to add the managed connector index prefix to the index name
+  const prefixConnectorIndex = (indexName: string) => {
+    if (!connector.is_native) {
+      return indexName;
+    }
+    if (indexName.startsWith(MANAGED_CONNECTOR_INDEX_PREFIX)) {
+      return indexName;
+    }
+    return `${MANAGED_CONNECTOR_INDEX_PREFIX}${indexName}`;
+  };
+
+  const [query, setQuery] = useState<{
+    isFullMatch: boolean;
+    searchValue: string;
+  }>();
+  const [sanitizedName, setSanitizedName] = useState<string>(
+    prefixConnectorIndex(formatApiName(connector.name))
+  );
+
   const [selectedIndex, setSelectedIndex] = useState<
     { label: string; shouldCreate?: boolean } | undefined
   >(
     connector.index_name
       ? {
-          label: connector.index_name,
+          label: removePrefixConnectorIndex(connector.index_name),
         }
       : undefined
   );
-  const [selectedLanguage] = useState<string>();
-  const [query, setQuery] = useState<{
-    isFullMatch: boolean;
-    searchValue: string;
-  }>();
-  const [sanitizedName, setSanitizedName] = useState<string>(formatApiName(connector.name));
-
-  const { makeRequest } = useActions(FetchAvailableIndicesAPILogic);
-  const { data, status } = useValues(FetchAvailableIndicesAPILogic);
-  const isLoading = [Status.IDLE, Status.LOADING].includes(status);
 
   const onSave = () => {
-    if (selectedIndex?.shouldCreate) {
-      createIndex({ indexName: selectedIndex.label, language: selectedLanguage ?? null });
-    } else if (selectedIndex && !(selectedIndex.label === connector.index_name)) {
-      attachIndex({ connectorId: connector.id, indexName: selectedIndex.label });
+    if (!selectedIndex) return;
+    const prefixedIndex = prefixConnectorIndex(selectedIndex.label);
+    if (selectedIndex.shouldCreate) {
+      createIndex({
+        indexName: prefixedIndex,
+        language: null,
+      });
+    } else if (connector.index_name !== prefixedIndex) {
+      attachIndex({
+        connectorId: connector.id,
+        indexName: prefixedIndex,
+      });
     }
   };
 
   const options: Array<EuiComboBoxOptionOption<string>> = isLoading
     ? []
-    : data?.indexNames.map((name) => {
-        return {
+    : data?.indexNames
+        .filter((name) => !connector.is_native || name.startsWith(MANAGED_CONNECTOR_INDEX_PREFIX))
+        .map((name) => ({
           label: name,
-        };
-      }) ?? [];
+          value: removePrefixConnectorIndex(name),
```

Comment:
This is needed for combobox suggestions in dropdown ...

Reply:
These comments are really helpful for reviewing. I think you should include them in the code as comments so future maintainers understand easier.